lib/demos/material/data_table_demo.dart

```diff
@@ -36,13 +51,23 @@ class _DataTableDemoState extends State<DataTableDemo> {
   ) {
     _dessertsDataSource._sort<T>(getField, ascending);
     setState(() {
-      _sortColumnIndex = columnIndex;
-      _sortAscending = ascending;
+      // [RestorableBool]'s value cannot be null, so -1 is used as a placeholder
```

Comment:
@goderbauer This is also an interesting problem, since `RestorableBool` does not take a null value, but some Flutter widgets use `null` to represent something meaningful.

Reply:
We can add an `RestorableBoolN` to the framework (and RestorableIntN, etc.) where the N stands for "this property is nullable". Ian and I discussed that as an option before. Feel free to open a PR on framework for that, if you need it.